src/vector/feature.rs

```diff
@@ -55,6 +71,52 @@ impl<'a> Feature<'a> {
         }
         return &self.geometry;
     }
+
+    pub fn create(&self, lyr: &Layer) -> Result<()> {
+        let rv = unsafe { ogr::OGR_L_CreateFeature(lyr.gdal_object_ptr(), self.c_feature) };
+        if rv != ogr_enums::OGRErr::OGRERR_NONE {
+            return Err(ErrorKind::OgrError(rv, "OGR_L_CreateFeature").into());
+        }
+        Ok(())
+    }
+
+    pub fn set_field_string(&self, field_name: &str, value: &str){
+        let c_str_field_name = CString::new(field_name).unwrap();
```

Comment:
Error handling in rust is a bit difficult. Where is a lot about [it in the book](https://doc.rust-lang.org/stable/book/error-handling.html).

To make this easier rust-gdal uses the [error-chain crate](https://docs.rs/error-chain/0.7.2/error_chain/). We already have some wrapping errors in errors.rs:
```
FfiNulError(::std::ffi::NulError);
StrUtf8Error(::std::str::Utf8Error);
```

To handle the `Ok `or return the (wrapped)  `Err` you can use the `try!{}` macro or `?`. In This case all you need to do is to replace `unwrap()` with `?`:  `let c_str_field_name = CString::new(field_name)?;` 
 `CString::new` returns `Result<CString, ::std::ffi::NulError>` in the error case `?` will convert `NulError` into the wrapping error (`FfiNulError`) and return it.